test/suites/integration/access_log_policy_test.go

```diff
@@ -567,25 +587,140 @@ var _ = Describe("Creating Access Log Policy", Ordered, func() {
 		}).Should(Succeed())
 	})
 
-	AfterEach(func() {
-		// TODO: Remove this block when DeleteAccessLogPolicy reconciliation is added
-		// Delete Access Log Subscriptions on test Service Network
-		listInput := &vpclattice.ListAccessLogSubscriptionsInput{
-			ResourceIdentifier: testServiceNetwork.Arn,
+	It("deletion removes the Access Log Subscription for the corresponding Service Network when the targetRef's Kind is Gateway", func() {
+		accessLogPolicy := &anv1alpha1.AccessLogPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      k8sResourceName,
+				Namespace: k8snamespace,
+			},
+			Spec: anv1alpha1.AccessLogPolicySpec{
+				DestinationArn: aws.String(bucketArn),
+				TargetRef: &gwv1alpha2.PolicyTargetReference{
+					Group:     gwv1beta1.GroupName,
+					Kind:      "Gateway",
+					Name:      gwv1alpha2.ObjectName(testGateway.Name),
+					Namespace: (*gwv1alpha2.Namespace)(aws.String(k8snamespace)),
+				},
+			},
 		}
-		output, err := testFramework.LatticeClient.ListAccessLogSubscriptionsWithContext(ctx, listInput)
-		Expect(err).To(BeNil())
-		for _, als := range output.Items {
-			deleteInput := &vpclattice.DeleteAccessLogSubscriptionInput{
-				AccessLogSubscriptionIdentifier: als.Arn,
-			}
-			_, err := testFramework.LatticeClient.DeleteAccessLogSubscriptionWithContext(ctx, deleteInput)
-			Expect(err).To(BeNil())
+		testFramework.ExpectCreated(ctx, accessLogPolicy)
+
+		Eventually(func(g Gomega) {
+			// Service Network should have an Access Log Subscription
+			output, err := testFramework.LatticeClient.ListAccessLogSubscriptions(&vpclattice.ListAccessLogSubscriptionsInput{
+				ResourceIdentifier: testServiceNetwork.Arn,
+			})
+			g.Expect(err).To(BeNil())
+			g.Expect(len(output.Items)).To(BeEquivalentTo(1))
+		}).Should(Succeed())
+
+		testFramework.ExpectDeleted(ctx, accessLogPolicy)
+
+		// Wait a moment for eventual consistency
+		time.Sleep(1 * time.Second)
```

Comment:
do you need sleep at all? it seemed to work fine in previous version with 1 nanosecond and can be removed

Reply:
The test will be flakey if we remove this. There is a <1 second expectation for eventual consistency to catch up on the List API used here